web/src/components/apps/AppLicense.tsx

```diff
@@ -65,36 +66,20 @@ const AppLicense = (props: Props) => {
     }
   );
 
-  const getAppLicense = async () => {
-    await fetch(`${process.env.API_ENDPOINT}/app/${props.app.slug}/license`, {
-      method: "GET",
-      headers: {
-        Authorization: Utilities.getToken(),
-        "Content-Type": "application/json",
-      },
-    })
-      .then(async (res) => {
-        const body = await res.json();
-        if (body === null) {
-          setState({ appLicense: null });
-        } else if (body.success) {
-          setState({
-            appLicense: body.license,
-            isViewingLicenseEntitlements:
-              size(body.license?.entitlements) <= 5 ? false : true,
-          });
-        } else if (body.error) {
-          console.log(body.error);
-        }
-      })
-      .catch((err) => {
-        console.log(err);
-      });
-  };
-
+  const { data: licenseWithInterceptResponse } = useLicenseWithIntercept();
   useEffect(() => {
-    getAppLicense();
-  }, []);
+    if (!licenseWithInterceptResponse) {
+      setState({ appLicense: null });
+    } else {
+      setState({
+        appLicense: licenseWithInterceptResponse.license,
+        isViewingLicenseEntitlements:
+          size(licenseWithInterceptResponse.license?.entitlements) <= 5
+            ? false
+            : true,
+      });
+    }
+  }, [licenseWithInterceptResponse]);
```

Comment:
- Removed the fetch license function and replace it with the new hook. 
- Added the response handling logic to the custom hook that watches for updates to the license response